yapapi/golem.py

```diff
@@ -59,6 +59,24 @@ class Golem(_Engine):
     at any given time.
     """
 
+    async def start(self) -> None:
+        """Initialize resources and start background services used by this engine.
+
+        Calling this method is not necessary, it will be called either way internally
+        when the engine is used for the first time.
+        """
+        if not self.operative:
+            await self._start()
```

Comment:
I've noticed that there's a difference between:
```python
async with Golem(...) as golem:
   do_stuff_with(golem)
```
and 
```python
golem = Golem(...)
await golem.start()
do_stuff_with(golem)
await golem.stop()
```
in case an exception is raised in `golem._start()`. 

With context manager, `golem._stop()` will be called before re-raising the exception. 

With the `start()/stop()` combo, the developer has to make sure that `stop()` is called when `start()` raises an exception. That's fine, if it's intentional and we document this clearly.

Reply:
I think that if the developer has *any* code that ensures `golem` is stopped when an exception is raised in `do_stuff_with(golem)` it will also work for start?

E.g. our "recommended" way of doing things without context manager would be

```
golem = Golem(...)
try:
    do_stuff_with(golem)
finally:
    golem.stop()
```
(I skipped 'start` because it is optional and I don't think we need it in our examples)

Reply:
@johny-b As I said, that's fine for me. But let's remember about this when creating documentation for `start()/stop()`.